src/ExperimentTable.js

```diff
@@ -0,0 +1,188 @@
+      : checkedArray.push(accession)
+    this.setState({checkedArray: checkedArray})
+  }
+
+  render() {
+    const {selectedSearch, selectedKingdom, checkedArray, selectedNumber, currentPage, searchedColumn, searchQuery, orderedColumn, ordering} = this.state
+    const {host, data, tableHeader, enableDownload} = this.props
+
+    const dataArray = selectedSearch ? this.sort(data).filter(data => data && Object.values(data).some(value => value.toString().toLowerCase().includes(selectedSearch.toLowerCase()))) :
+      this.filter(this.sort(data), tableHeader).filter(data => selectedKingdom ? data.kingdom === selectedKingdom : true)
+    const currentPageData = selectedNumber ? dataArray.slice(selectedNumber*(currentPage-1), selectedNumber*currentPage) : dataArray
+
+    return [
+      <div key={`tableHead`} className={`row expanded`}>
+        <div className={`large-2 medium-4 small-8 columns`}>
+          <label> Kingdom:
+            <select defaultValue={``} className={`kingdom`}
+              onChange={event => this.setState({selectedKingdom: event.target.value})}>
+              <option value={``} >All</option>
+              <option value={`animals`}>Plants</option>
+              <option value={`plants`}>Animals</option>
+              <option value={`fungi`}>Fungi</option>
+            </select>
+          </label>
+        </div>
+        <div className={`large-2 medium-4 small-8 columns`}>
+          <label>Entries per page:
+            <select defaultValue={10} onChange={this.setValue(`selectedNumber`)}>
+              <option value={10}>10</option>
+              <option value={25}>25</option>
+              <option value={50}>50</option>
+              <option value={100}>100</option>
+              <option value={data.length}>All</option>
+            </select>
+          </label>
+        </div>
+        <div className={`large-2 medium-4 small-8 columns`}>
+          <label>Search all columns:
+            <input type={`search`} placeholder={`Type here ...`}
+              onChange={this.setValue(`selectedSearch`)}/></label>
+        </div>
+      </div>,
+
+      <Table border key={`table`} style={{display:`grid`}}>
+        <Table.Head>
+          <Table.TextHeaderCell key={`index`} flexBasis={100} flexShrink={0} flexGrow={0}>Index</Table.TextHeaderCell>
+          {[
+            <TableHeaderCells key={`tableHeader`} {...{tableHeader, searchedColumn, searchQuery, orderedColumn, ordering}}
+              onClick={(columnNumber) =>
+                this.setState({
+                  orderedColumn: columnNumber,
+                  ordering: !this.state.ordering})}
+              onChange={(value, columnNumber) =>
+                this.setState({
+                  searchQuery: value,
+                  searchedColumn: columnNumber
+                })}
+            />,
+            enableDownload && <Table.TextHeaderCell className={`downloadHeader`} key={`download`}>
```

Comment:
You have a lot of unnecessary keys on your elements. You should remove them. Read up on keys in React [here](https://reactjs.org/docs/lists-and-keys.html#keys).

Reply:
Right. I used an array as the `return` object to avoid wrapper them in an extra `div`, so it required those keys, which is over-complicated. I have remove them.